StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Controllers/PayWithLinkWebController.swift

```diff
@@ -159,21 +159,42 @@ final class PayWithLinkWebController: NSObject, ASWebAuthenticationPresentationC
     }
 
     var webAuthSession: ASWebAuthenticationSession?
+
     /// Defaults to the app's key window
-    func present(over viewController: UIViewController) {
-//      TODO: Log attempt to show web session
-        let webAuthSession = ASWebAuthenticationSession(url: LinkURLGenerator.url(), callbackURLScheme: "stripesdk") { url, _ in
-            guard let url = url else {
-                //                  TODO: Get analytics logs here from error: Did the user start a session and then reject the non-ephemeralSession dialog? ASWebAuthenticationSession.cancelledLogin error
-                return
+    func present(over viewController: UIViewController? = nil) {
+        Task { @MainActor in
+            // TODO: Log attempt to show web session
+            do {
+                let url = try await LinkURLGenerator.url(configuration: self.context.configuration, intent: self.context.intent)
+                let webAuthSession = ASWebAuthenticationSession(url: url, callbackURLScheme: "link-popup") { [self] url, error in
```

Comment:
It's a bit confusing to use the same name of `url` for both the linkURLGenerator and the url that is being returned in the completion block of ASWebAuthenticationSession.

maybe something like:
```
let generatedURL = try await LinkURLGenerator....
let webAuthSession = ASWebAuthenticationSession(url: generatedUrl,...)
```